src/Splitio/Services/InputValidation/Classes/ApiKeyValidator.cs

```diff
@@ -18,11 +19,13 @@ public void Validate(string apiKey)
             if (apiKey == string.Empty)
             {
                 _log.Error("factory instantiation: you passed and empty api_key, api_key must be a non-empty string.");
+                throw new ArgumentException("API Key must be set to initialize Split SDK.", nameof(apiKey));
```

Comment:
```suggestion
                throw new Exception("API Key should be set to initialize Split SDK.");
```

Reply:
we can not change the Exception type.

Reply:
Sonar complained we are using too generic exception